lambdas/cloudwatch_logs/main.js

```diff
@@ -0,0 +1,239 @@
+// Based on AWS Cloudwatch "Stream to Amazon Elasticsearch Service" funcion v1.1.2
+// Required stack parameters:
+// * EnvVarElasticsearchEndpoint - Amazon Elasticsearch Service domain endpoint
+
+var https = require('https');
+var zlib = require('zlib');
+var crypto = require('crypto');
+
+var environment = JSON.parse(require('fs').readFileSync('env', 'utf8'));
+var endpoint = environment['ELASTICSEARCH_ENDPOINT'];
+
+exports.handler = function(input, context) {
+    // decode input from base64
+    var zippedInput = new Buffer(input.awslogs.data, 'base64');
+
+    // decompress the input
+    zlib.gunzip(zippedInput, function(error, buffer) {
+        if (error) { context.fail(error); return; }
+
+        // parse the input from JSON
+        var awslogsData = JSON.parse(buffer.toString('utf8'));
+
+        // transform the input to Elasticsearch documents
+        var elasticsearchBulkData = transform(awslogsData);
+
+        // skip control messages
+        if (!elasticsearchBulkData) {
+            console.log('Received a control message');
```

Comment:
It creates a new log group in CloudWatch for each function. It's a bit annoying since I don't think there's a way to configure it and it sets them to never expire by default. But I don't think the functions write very much, and I set a shorter expiration for preview manually.